### ISSUE TITLE
[Bug] Worker started state is not reset after shutdown

### ISSUE THREAD

**Issue by schibrikov** [temporalio/sdk-python#286]
### What are you really trying to do?

I want to shutdown worker and then spin it up again later. So I'm calling `worker.shutdown()`, and then after it is off for some time calling `worker.run()` again.

### Describe the bug

`run()` throws `Already started` error, despite worker has shut down already. This happens because `self._started` is set to `True` when starting, but not reset when shutting down.

I believe this is a bug and we should reset `self._started` flag to False, when worker has shut down.
Is it the case or I'm doing something very wrong?

### Extra context

My worker is processing long-running, resource-consuming tasks, so that only 1 can be processed at a time. Though the whole workflow consists of 3 different tasks (file-system dependent) that should run on a single node. So effectively I'm running 3 workers:

1. One is for the workflow itself, no activities
2. Second contains an activity that reports unique worker id and that shut down itself
3. Third processes all 3 long-running activities on the unique task queue and after that, in a separate activity - restarts the second worker.

Is this scheme meaningful or I can get it running simply other way?

**Comment by cretz** [comment]
>  Is it the case or I'm doing something very wrong?

Workers are single use by intention to prevent confusion and misuse. You can create another worker with the same config and call `run` on it.

> Is this scheme meaningful or I can get it running simply other way?

I would need to know more about _why_ you need to restart a worker. Most workers run forever and you use workflows to orchestrate or limit activities. If easier this can be discussed in our forum or Slack.

**Comment by schibrikov** [comment]
Yeah, right, I had no luck manually setting `worker._started = False` after shut down, so there are other concerns with restarting it. And yes, I got around this issue by simply running a new worker, instead of reusing, so they kind of "one-off" now and it works exactly as I expected, thanks.

I would happily describe my use-case in details later, since it appears common to me and may worth documenting.
In short: I'm running ML workflow (download-inference-upload/cleanup). So I need to stick my workflow to a single node and run a single one on this node, also inference is long-running, so I don't want to schedule the next task to a node currently computing. And I cannot simply follow sticky sessions tutorial, since my set of nodes is dynamic.

**Comment by cretz** [comment]
> I would happily describe my use-case in details later

:+1: Will be on the look out and close this issue as not a bug.

> And I cannot simply follow sticky sessions tutorial, since my set of nodes is dynamic.

The sticky sessions example at https://github.com/temporalio/samples-python/tree/main/activity_sticky_queues is for a dynamic number of nodes too. You don't have to use UUID to identify each host if you need more specific service discovery. You can also use workflows identified by host-specific identifiers to orchestrate activities onto host-specific task queues if needed.

Can discuss in another place if needed, closing as not a bug.
